Migrate/unit staff editor

atharv02-git opened the pull request:
# Description

This is a pull request for the migration of the UnitStaffEditor component. This component focusses on the modifying the unit staff thereby letting main convenor adding staff members to the unit, and assigning them a tutor or convenor role.
It has been under the migration from the coffee script  and Angular to Type Script and Angular17. The html file uses a `mat-autocomplete` feature from the Angular Material. It functions the same as before, however there should be some changes still needing at the html side. There are no any external dependencies required for the component to be run.


## Type of change


- [ ] Migration

# How Has This Been Tested?

- Login to the OnTrack as **aconvenor** and go to **http://localhost:4200/#/units/3/admin**.
- When you login as "aconvenor" and you test the functionality of the **change role** button you will be notified as **permission errors** from backend so I will recommend you, to login as "aadmin" and check the funtionality of the component.
- There you can see list of staff members and a input box below where you can add new staff, change role, or delete the staff from the list.
- Once done, you can refresh the page to see if the changes are reflected or not.

## Before
![Before](https://github.com/user-attachments/assets/a16b632c-6dae-4bc3-9378-37106121fc7f)

## After
![role changed](https://github.com/user-attachments/assets/a40744b0-73e6-44c3-a15b-35476a49f7e2)


## Testing Checklist:

- [✅] Tested in latest Chrome
- [✅] Tested in latest Safari
- [✅] Tested in latest Firefox

# Checklist:

- [✅] My code follows the style guidelines of this project
- [✅] I have performed a self-review of my own code
- [✅] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [✅] My changes generate no new warnings

atharv02-git committed:
New: Create initial files for migration of unit-staff-editor

atharv02-git committed:
New: Create initial files for migration of unit-staff-editor

atharv02-git committed:
New: Create initial files for migration of unit-staff-editor

atharv02-git committed:
New: Create initial files for migration of unit-staff-editor

atharv02-git committed:
New: Create initial files for migration of unit-staff-editor

atharv02-git committed:
NEW: Create initial files for migration of unit-staff-editor

atharv02-git committed:
NEW: Create initial files for migration of unit-staff-editor

AusBruce commented:
have you made a pull request to doubutfire-lms?

atharv02-git commented:
> have you made a pull request to doubutfire-lms?

Hi I haven't made a pull request to doubtfire-lms, do we have to make a pull request to the "doubtfire-web lms version" or the "doubtfire-web thothtech version"?

JackSCarroll commented:
This looks like a good start but there still some things you need to work on. I'll list my thoughts for each section:

**Name section**

- Displays names correctly with usernames. 😊
- Icons displayed nicely. 😊
- Consider moving the "Name" title and staff names close to the icons to more closely match the original component. 🤔

**Role section**

- The grey highlight is raised in your version but it is sunken in the original. 🛠️
- Could probably remove the tick icon and just have a 50/50 split. 🤔
- Selecting one of the options causes the whole element to jump around. 🛠️
- The functionality of the button doesn't work, I get errors about unable to find staff id and the changes don't persist after  refreshing the page. 🛠️

**Main Convenor section**

- The tick icon is very small and low contrast. Increasing the size and adding a border would help with this. 🛠️
- Also the original has a cross for those that aren't convenors, should add this too. 🛠️

**Actions section**

- This matches the styling of the existing task creation component. 😊
- Functions as expected. 😊

**Add Staff section**

- Needs padding on the left and right so it doesn't touch the borders. 🛠️
- Man icon and Plus icon are too small. 🛠️
- Name search works as expected and adding staff to unit staff works and persists 😊

b0ink commented:
While looking at the behaviour of the original component, I noticed that if you're logged in as `aconvenor`, you have the permissions to assign yourself back to Tutor, however this revokes your permission to be able to reassign yourself (and other tutors) between tutor/convenor

The functionality of the `changeRole()` function still needs be fixed, but you will still encounter similar permission issues, so to test this properly I'd recommend logging in as an `aadmin` to have full permissions

atharv02-git committed:
Delete src/app/units/states/edit/directives/unit-staff-editor/unit-staff-editor.coffee

deleted unit-staff-editor.coffee file

atharv02-git committed:
Delete src/app/units/states/edit/directives/unit-staff-editor/unit-staff-editor.tpl.html

Deleted unit-staff-editor.tpl.html file

atharv02-git committed:
Update unit-staff-editor.component.html

Fix Role Change Button and HTML styling

atharv02-git committed:
Update unit-staff-editor.component.scss

atharv02-git committed:
Update unit-staff-editor.component.ts

Fixed change role button and html styling

atharv02-git commented:
### This commit addresses the feedback from Jack and Eliya's reviews.

1. The button functionality has been fixed, and now changes persist even after refreshing the page. 

- Note: I recommend logging in as aadmin instead of aconvenor when testing functionality. Logging in as aconvenor allows you to change your role from convenor to tutor, but you won’t be able to reassign yourself as convenor afterward, resulting in an API error.

2. The tick icon couldn't be removed since it's part of the mat-button-toggle group.
3.  A cross icon has been added for users who are not convenors.
4. After reviewing other migrated components, I noticed that the "Add" button uses the primary color instead of green, so I updated it to match.
5. I added padding on the left and right to prevent the section from touching the borders.
6. The Man and Plus icons have been removed.
